pragma-node/src/handlers/entries/mod.rs

```diff
@@ -77,22 +78,59 @@ pub struct GetVolatilityResponse {
 pub struct GetOnchainParams {
     pub network: Network,
     pub aggregation: Option<AggregationMode>,
-    pub timestamp: Option<u64>,
     pub routing: Option<bool>,
+    #[serde(default, with = "timestamp_serde")]
+    pub timestamp: Option<Timestamp>,
+}
+
+#[derive(Debug, Clone)]
```

Comment:
Also this `Timestamp` type should probably live in the `types` module

Reply:
Done

Reply:
It should not be in `pragma-common` but here imo: https://github.com/astraly-labs/pragma-node/tree/main/pragma-node/src/types

since it's only used for endpoint processing

Reply:
oh yes sry

Reply:
should be good now